libqtile/backend/wayland/core.py

```diff
@@ -300,6 +307,18 @@ def get_screen_for_output(self, output: ffi.CData) -> Screen:
 
         return self.qtile.current_screen
 
+    def handle_get_current_output_dims(self) -> ffi.CData:
+        assert self.qtile is not None
+
+        output_dims = ffi.new("struct wlr_box *")
```

Comment:
@jwijenbergh @richcarni For my own education, how does this data get tidied up?

Reply:
by the python GC/refcounter. This function creates a struct and returns it (so the data is copied over). You can see when it is cleaned up using (untested):

```python
output_dims = ffi.new("struct wlr_box *")
output_dims = ffi.gc(output_dims, lambda p: print("output dims are cleaned up"))
```